@here/harp-mapview/lib/MapView.ts

```diff
@@ -1261,6 +1269,12 @@ export class MapView extends THREE.EventDispatcher {
      * cleanup, you must ensure that all references to this `MapView` are removed.
      */
     dispose() {
+        this.m_disposed = true;
+
+        // Enforce listeners that we are about to dispose.
+        DISPOSE_EVENT.time = Date.now();
+        this.dispatchEvent(DISPOSE_EVENT);
```

Comment:
Since the event is fired before the actual disposal happens I would also move `this.m_disposed = true;` below the dispatch event call

Reply:
OK...